archive_updater_test.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import subprocess
+import os
+import tempfile
+import shutil
+from archive_updater import fill_message_directory
+from unittest import mock
+
+class ArchiveUpdaterFillMessageDirectoryTest(unittest.TestCase):
+
+    def setUp(self):
+        self.tmp_dir = tempfile.mkdtemp()
+    def tearDown(self):
+        shutil.rmtree(self.tmp_dir)
+
+    @mock.patch.object(subprocess, 'check_call')
+    def test_update_failure(self, mock_check_call):
+        mock_check_call.side_effect = subprocess.CalledProcessError('Failed to fetch', 'git fetch')
+        self.assertRaises(subprocess.CalledProcessError,
+                          fill_message_directory, '../linux-kselftest/git/0.git',
+                                                 '../lkml-gerrit-bridge/test_data',
+                                                 '5227e860278407173cbfdab594676764153c27e1')
+
+    @mock.patch.object(subprocess, 'check_output')
+    @mock.patch.object(subprocess, 'call')
+    @mock.patch.object(subprocess, 'check_call')
+    def test_fill_message_directory_success(self, mock_check_call, mock_call, mock_check_output):
+        def writeFile(args, stdout):
+            stdout.write('called with: ' + ' '.join(args))
+        mock_check_call.return_value = ''
+        mock_call.side_effect = writeFile
+        mock_check_output.return_value = b'success1 success2 success3'
+
+        last_used_hash = fill_message_directory('archive_path', self.tmp_dir, '')
+
+        self.assertEqual(len(os.listdir(self.tmp_dir)), 3)
+        for filename in os.listdir(self.tmp_dir):
+            with open(os.path.join(self.tmp_dir,filename), "r") as file:
+                val = file.read()
+                self.assertIn('git -C archive_path show', val)
+        self.assertEqual(last_used_hash, 'success1')
+
+    @mock.patch.object(subprocess, 'check_output')
+    @mock.patch.object(subprocess, 'check_call')
+    def test_fill_message_directory_fail_to_log(self, mock_check_call, mock_check_output):
+        mock_check_call.return_value = ''
+        mock_check_output.side_effect = subprocess.CalledProcessError('Failed to log', 'git log')
```

Comment:
I don't think this is initializing the exception like you intended.
You're setting the `returncode` to 'Failed to log' in this case.

Do you want to use named parameters here to make it more clear?